File: 101-symmetric-tree/101-symmetric-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public boolean isSymmetric(TreeNode root) {
        if(root == null) {
            return false;
        }
        return check(root.left,root.right);
    }
    public boolean check(TreeNode l, TreeNode r) {
        if(l == null && r == null) {
            return true;
        }
        if(l == null || r == null) {
            return l == r ? true:false;
        }
        if(l.val != r.val) {
            return false;
        }

        return check(l.left,r.right)&&check(l.right,r.left);
    } 
}

File: 101-symmetric-tree/101-symmetric-tree.go
// https://leetcode.cn/problems/symmetric-tree/
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isSymmetric(root *TreeNode) bool {
    if root == nil {
        return true
    }
    return check(root.Left,root.Right)
    //下面判断两边的二叉树是否相等

}
func check(Left *TreeNode, Right *TreeNode) bool{
        if Left==nil || Right == nil {
            return (Left==Right)
        }
        if Left==nil && Right==nil {
            return true
        }
        if Left.Val != Right.Val {
            //如果左右子树的值不相同
            return false
        }
        return check(Left.Right, Right.Left)&&check(Left.Left,Right.Right)
    }